src/NativeScript/inspector/GlobalObjectConsoleClient.cpp

```diff
@@ -0,0 +1,151 @@
+    const Inspector::ScriptCallFrame& lastCaller = callStack->at(0);
+
+    StringBuilder builder;
+
+    if (!lastCaller.sourceURL().isEmpty()) {
+        appendURLAndPosition(builder, lastCaller.sourceURL(), lastCaller.lineNumber(), lastCaller.columnNumber());
+        builder.appendLiteral(": ");
+    }
+
+    if (type == JSC::MessageType::Dir) {
+        builder.append(this->getDirMessage(exec, arguments));
+    } else {
+        for (size_t i = 0; i < arguments->argumentCount(); ++i) {
+            String argAsString = arguments->argumentAt(i).toString(arguments->globalState());
+            builder.append(' ');
+            builder.append(argAsString);
+        }
+    }
+
+    ConsoleClient::printConsoleMessage(source, type, level, builder.toString(), WTF::emptyString(), 0, 0);
+}
+
+void GlobalObjectConsoleClient::count(JSC::ExecState* exec, RefPtr<Inspector::ScriptArguments>&& arguments) {
+    m_consoleAgent->count(exec, arguments);
+}
+
+void GlobalObjectConsoleClient::profile(JSC::ExecState* execState, const String&) {
+    NativeScript::GlobalObject* globalObject = JSC::jsCast<GlobalObject*>(execState->lexicalGlobalObject());
+    Inspector::InspectorTimelineAgent* timelineAgent = globalObject->inspectorController().timelineAgent();
+    if (timelineAgent) {
+        Inspector::ErrorString unused;
+        timelineAgent->start(unused, nullptr);
+    }
+}
+
+void GlobalObjectConsoleClient::profileEnd(JSC::ExecState* execState, const String&) {
+    NativeScript::GlobalObject* globalObject = JSC::jsCast<NativeScript::GlobalObject*>(execState->lexicalGlobalObject());
+    Inspector::InspectorTimelineAgent* timelineAgent = globalObject->inspectorController().timelineAgent();
+    if (timelineAgent) {
+        Inspector::ErrorString unused;
+        timelineAgent->stop(unused);
+    }
+}
+
+void GlobalObjectConsoleClient::time(JSC::ExecState*, const String& title) {
+    m_consoleAgent->startTiming(title);
+}
+
+void GlobalObjectConsoleClient::timeEnd(JSC::ExecState* exec, const String& title) {
+    RefPtr<Inspector::ScriptCallStack> callStack(Inspector::createScriptCallStackForConsole(exec, 1));
+    m_consoleAgent->stopTiming(title, WTF::move(callStack));
+}
+
+void GlobalObjectConsoleClient::timeStamp(JSC::ExecState*, RefPtr<Inspector::ScriptArguments>&&) {
+    // FIXME: JSContext inspection needs a timeline.
+    warnUnimplemented(ASCIILiteral("console.timeStamp"));
+}
+
+void GlobalObjectConsoleClient::warnUnimplemented(const String& method) {
+    String message = method + " is currently ignored in JavaScript context inspection.";
+    m_consoleAgent->addMessageToConsole(std::make_unique<Inspector::ConsoleMessage>(MessageSource::ConsoleAPI, MessageType::Log, MessageLevel::Warning, message, nullptr, nullptr));
+}
+
+WTF::String GlobalObjectConsoleClient::getDirMessage(JSC::ExecState* exec, RefPtr<Inspector::ScriptArguments>& arguments) {
```

Comment:
Since this method always takes the first argument, shouldn't it just be declared as accepting a single `JSValue` instead of a `ScriptArguments` object?